src/engraving/dom/tie.h

```diff
@@ -36,13 +36,20 @@ class TieSegment final : public SlurTieSegment
     OBJECT_ALLOCATOR(engraving, TieSegment)
     DECLARE_CLASSOF(ElementType::TIE_SEGMENT)
 
-    PointF autoAdjustOffset;
+    double m_midThickness = 0.0;
+
+    std::array<PointF, static_cast<size_t>(Grip::GRIPS)> m_adjustOffsets = { PointF() };
```

Comment:
This does not have the desired effect. You're probably thinking of `std::vector`'s constructor, but `std::array` does not have a similar thing. It only has "aggregate initialisation", meaning that you'll have essentially the same effect as with plain old C arrays:
```c++
    … m_adjustOffsets = { 
        PointF(),
        <default_initialised_by_compiler>,
        <default_initialised_by_compiler>,
        ⋮
        <default_initialised_by_compiler> // static_cast<size_t>(Grip::GRIPS) - 1 times
    };
```
Luckily, for structs and classes, `<default_initialised_by_compiler>` will be `PointF()`, so it will work correctly. But that means you could also not have specified anything at all, so that the first entry is also `<default_initialised_by_compiler>`; I'd argue that it's better to do that.
```suggestion
    std::array<PointF, static_cast<size_t>(Grip::GRIPS)> m_adjustOffsets;
```